### PR TITLE
Auto basemap generation if tms url is input during project creation

### PR DESCRIPTION
**Pull request by Anuj-Gupta4**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Related Issue

- [x] Fixes: #2025 
- [x] Resolves: #1953 

## Describe this PR

- Added extra option to tile source of BasemapGenerate pydantic model.
- Added tms basemap generation during project data generation if tms url was present during project creation.

### CODE REVIEW

File: src/backend/app/projects/project_routes.py [hunk 1]
```diff
@@ -942,6 +944,13 @@ async def generate_files(
             },
         )
 
+    if project.custom_tms_url:
+        basemap_in = project_schemas.BasemapGenerate(
+            tile_source="tms", file_format="pmtiles", tms_url=project.custom_tms_url
+        )
+        org_id = project.organisation_id
+        await generate_basemap(project_id, org_id, basemap_in, db, background_tasks)
```
Comment (spwoodcock) [inline src/backend/app/projects/project_routes.py:952]: Its key that this returns immediately when the background task is submitted - just checking this works well upon testing?

**Review by spwoodcock — APPROVED** [review]
This is great to have in place 😄

In theory this should as minimal time to the project file creation API response time.

So the frontend will redirect and the project can be viewed. The basemap generation will lock a thread, but there should generally be another thread available to serve the project response afterwards.

This will have more guarantees in future with a separate [basemap-api](https://github.com/hotosm/basemap-api) microservice though

File: src/backend/app/projects/project_schemas.py [hunk 2]
```diff
@@ -274,7 +274,9 @@ class ProjectUserContributions(BaseModel):
 class BasemapGenerate(BaseModel):
     """Params to generate a new basemap."""
 
-    tile_source: Annotated[Literal["esri", "bing", "google"], Field(default="esri")]
+    tile_source: Annotated[
+        Literal["esri", "bing", "google", "tms"], Field(default="esri")
```
Comment (spwoodcock) [inline src/backend/app/projects/project_schemas.py:278]: Good catch 👍

File: src/backend/app/projects/project_routes.py (hunk 1 above)
Comment (Anuj-Gupta4) [reply src/backend/app/projects/project_routes.py:952]: Yes this worked upon testing

**Review by Sujanadh — APPROVED** [review]

**Pull request merged**